docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,604 @@
+not mean it actually writes that type. In the example above, the writer might
+accept a set of layers containing `image`s and `point`s, but the write command
+might just ignore the `point` layers. The writer must return `None` for
+unwritten layers.
+
+### Calling convention
+
+Currently, two calling conventions are supported for writers: single-layer and
+multi-layer writers. When at most one layer can be matched by a writer, it
+must use the single-layer convention. Otherwise, the multi-layer convention
+must be used.
+
+#### multi-layer writer
+
+```python
+def writer_function(
+    path: str, layer_data: List[LayerData]
+) -> List[str]:
+    ...
+```
+
+**Parameters**
+
+path(str)
+: Path to file, directory, or resource (like a url).
+
+layer_data(list of LayerData)
+: Each `LayerData` element is a tuple of `(data,meta,layer_type)`.
+
+**Return**
+
+Returns a list of paths that were successfully written.
+
+#### single-layer writer
+
+```python
+def writer_function(
+    path: str, data, meta
+) -> Optional[str]:
+    ...
+```
+
+**Parameters**
+
+path(str)
+: Path to file, directory, or resource (like a url).
+
+data(array or list of array) : Layer data. Can be N dimensional. If an image
+and meta[‘rgb’] is True then the data should be interpreted as RGB or RGBA. If
+meta[‘multiscale’] is True, then the data should be interpreted as a
+multiscale image.
+
+meta(dict)
+: Layer metadata.
+
+**Return**
+
+If data is successfully written, return the path that was written. Otherwise,
+if nothing was done, return None.
+
+## Sample Data
+
+There are two ways of specifying sample data.
+
+1. As a **sample data generator**, a function that returns layer data.
+2. As a **sample data uri**, a `uri` that should be read using a `reader`
+   plugin.
+
+### Required fields
+
+_sample data generator_
+
+- **command** Identifier of a command that returns a layer data tuple.
+- **display_name** String to show in the GUI when referring to this sample.
+- **key** A key to identify this sample. Must be unique across the samples
+  provided by a single plugin.
+
+_sample data uri_
+
+- **uri** Path or URL to a data resource.
+- **display_name** String to show in the GUI when referring to this sample.
+- **key** A key to identify this sample. Must be unique across the samples
+  provided by a single plugin.
+
+### Optional fields
+
+_sample data uri_
+
+- **reader_plugin** Name of plugin to use to open the `uri`.
+
+### Examples
+
+**Example**
+
+Sample data generator
+
+```yaml
+name: napari
+contributions:
+  commands:
+    - id: napari.data.astronaut
+      title: Generate astronaut sample
+      python_name: napari.plugins._skimage_data:astronaut
+  sample_data:
+    - display_name: Astronaut (RGB)
+      key: astronaut
+      command: napari.data.astronaut
+```
+
+**Example**
+
+Sample data uri
+
+```yaml
+name: my-sample
+contributions:
+  sample_data:
+    - display_name: Tabueran Kribati
```

Comment:
I'm just browsing for now (no full review yet!) here, but my typo sensor picked this one out.
```suggestion
    - display_name: Tabueran Kiribati
```